mapper/orm-coordination-outbox-polling/src/main/java/org/hibernate/search/mapper/orm/coordination/outboxpolling/cluster/impl/OutboxPollingAgentAdditionalJaxbMappingProducer.java

```diff
@@ -168,38 +132,55 @@ public Collection<MappingDocument> produceMappings(ConfigurationPropertySource p
-						.strategy(),
-				resolvedUuidType,
-				PayloadMappingUtils.payload(
-						payloadType.orElse(
-								HibernateOrmMapperOutboxPollingSettings.Defaults.COORDINATION_ENTITY_MAPPING_AGENT_PAYLOAD_TYPE ),
-						true
-				)
-		) );
-
-		log.agentGeneratedEntityMapping( entityDefinition );
-		Origin origin = new Origin( SourceType.OTHER, "search" );
-
-		ByteArrayInputStream byteArrayInputStream = new ByteArrayInputStream( entityDefinition.getBytes() );
-		BufferedInputStream bufferedInputStream = new BufferedInputStream( byteArrayInputStream );
-		Binding<?> binding = mappingBinder.bind( bufferedInputStream, origin );
-
-		JaxbHbmHibernateMapping root = (JaxbHbmHibernateMapping) binding.getRoot();
-
-		MappingDocument mappingDocument = new MappingDocument( HIBERNATE_SEARCH, root, origin, buildingContext );
-		return Collections.singletonList( mappingDocument );
+		JaxbEntityMappings mappings;
+		if ( mapping.isPresent() ) {
+			mappings = unmarshall( mapping.get() );
+		}
+		else {
+			int resolvedUuidType = UuidDataTypeUtils.uuidType(
+					uuidType.orElse(
+							HibernateOrmMapperOutboxPollingSettings.Defaults.COORDINATION_ENTITY_MAPPING_AGENT_UUID_TYPE ),
+					dialect );
+			@SuppressWarnings("deprecation")
+			int resolvedPayloadType = PayloadMappingUtils.payload(
+					payloadType.orElse(
+							HibernateOrmMapperOutboxPollingSettings.Defaults.COORDINATION_ENTITY_MAPPING_AGENT_PAYLOAD_TYPE )
+			);
+			String resolvedUuidStrategy = uuidStrategy.orElse(
+					HibernateOrmMapperOutboxPollingSettings.Defaults.COORDINATION_ENTITY_MAPPING_AGENT_UUID_GEN_STRATEGY )
+					.strategy();
+
+			mappings = createMappings(
+					schema.orElse( "" ),
+					catalog.orElse( "" ),
+					table.orElse( HibernateOrmMapperOutboxPollingSettings.Defaults.COORDINATION_ENTITY_MAPPING_AGENT_TABLE ),
+					resolvedUuidType, resolvedPayloadType, resolvedUuidStrategy,
+					buildingContext.getMetadataCollector().getFilterDefinition( TenantIdBinder.FILTER_NAME ) != null
+			);
+		}
+
+		log.agentGeneratedEntityMapping( marshall( mappings ) );
+
+		return Map.of( Agent.class, mappings );
+	}
+
+	private static JaxbEntityMappings createMappings(String schema, String catalog,
+			String table, int resolvedUuidType, int resolvedPayloadType, String resolvedUuidStrategy,
+			boolean tenantIdRequired) {
+		AdditionalMappingContributorBuilder builder = new AdditionalMappingContributorBuilder(
+				Agent.class, ENTITY_NAME )
+				.id( resolvedUuidType, resolvedUuidStrategy )
+				.table( schema, catalog, table )
+				.enumAttribute( "type", null, false )
+				.attribute( "name", null, false )
+				.attribute( "expiration", null, false )
+				.enumAttribute( "state", null, false )
+				.attribute( "totalShardCount", null, true )
+				.attribute( "assignedShardIndex", null, true )
+				.attribute( "payload", Length.LONG32, true, resolvedPayloadType );
+		if ( tenantIdRequired ) {
+			builder.tenantId( "tenantId" );
+		}
```

Comment:
Did you test somehow that this is effective, i.e. Hibernate ORM doesn't expect a `tenantId` column in the table when using single-tenancy or schema-based/database-based multi-tenancy?